### ISSUE TITLE
Expose bcf_get_value_count on the record object

### ISSUE THREAD

**Issue by TMiguelT** [pysam-developers/pysam#669]
`bcf_get_value_count` seems to calculate how many values a given INFO/FMT field should have, on a given record. This is a useful feature, if you want to edit a VCF record but don't know how many values each field should have. Could this be added to the `VariantRecord` object, e.g. as `VariantRecord#get_value_count(type: str, id: str)`?

I'll probably try writing this myself at some point, but I wanted to keep track of this as an issue

**Comment by bioinformed** [comment]
Good idea!  It will be a few weeks before I have time to work on this, but it will be extremely useful.

**Comment by TMiguelT** [comment]
Any chance of merging this?

**Comment by TMiguelT** [comment]
Totally agree with the interface of putting `value_count` inside the sample/info object. I forgot that they were actually C objects and not just dictionaries, and so can have methods.

I'm not sure I follow in terms of `GT` though. It's a field that always has a Number of R, effectively (although `|` or `/` separated, rather than comma separated). You can calculate the length of the field with `len(gt)`, but I see no reason not to allow this function to cover all fields in a general way.
